File: src/2025-03-04-03-28-40--deepseek-chat-0304/go/exercises/practice/beer-song/beer_song.go
package beer

import (
	"errors"
	"fmt"
	"strings"
)

func Verse(n int) (string, error) {
	if n < 0 || n > 99 {
		return "", errors.New("invalid verse number")
	}

	switch n {
	case 0:
		return "No more bottles of beer on the wall, no more bottles of beer.\n" +
			"Go to the store and buy some more, 99 bottles of beer on the wall.\n", nil
	case 1:
		return "1 bottle of beer on the wall, 1 bottle of beer.\n" +
			"Take it down and pass it around, no more bottles of beer on the wall.\n", nil
	case 2:
		return "2 bottles of beer on the wall, 2 bottles of beer.\n" +
			"Take one down and pass it around, 1 bottle of beer on the wall.\n", nil
	default:
		return fmt.Sprintf("%d bottles of beer on the wall, %d bottles of beer.\n"+
			"Take one down and pass it around, %d bottles of beer on the wall.\n",
			n, n, n-1), nil
	}
}

func Verses(start, stop int) (string, error) {
	if start < stop || start < 0 || stop < 0 || start > 99 || stop > 99 {
		return "", errors.New("invalid verse range")
	}

	var verses []string
	for i := start; i >= stop; i-- {
		v, err := Verse(i)
		if err != nil {
			return "", err
		}
		verses = append(verses, v)
	}
	return strings.Join(verses, "\n") + "\n", nil
}

func Song() string {
	song, _ := Verses(99, 0)
	return song
}

File: src/2025-03-16-12-54-49--a-deepclaude-full-architect-python-0316-2/python/exercises/practice/beer-song/beer_song.py
def recite(start, take=1):
    """
    Generate the lyrics for the '99 Bottles of Beer on the Wall' song.
    
    Args:
        start: The number of bottles to start with
        take: The number of verses to generate
    
    Returns:
        A list of strings representing the lyrics
    """
    result = []
    
    for i in range(start, start - take, -1):
        # First line of verse
        if i == 0:
            result.append("No more bottles of beer on the wall, no more bottles of beer.")
        elif i == 1:
            result.append("1 bottle of beer on the wall, 1 bottle of beer.")
        else:
            result.append(f"{i} bottles of beer on the wall, {i} bottles of beer.")
        
        # Second line of verse
        if i == 0:
            result.append("Go to the store and buy some more, 99 bottles of beer on the wall.")
        elif i == 1:
            result.append("Take it down and pass it around, no more bottles of beer on the wall.")
        elif i == 2:
            result.append("Take one down and pass it around, 1 bottle of beer on the wall.")
        else:
            result.append(f"Take one down and pass it around, {i-1} bottles of beer on the wall.")
        
        # Add blank line between verses except after the last verse
        if i > start - take + 1:
            result.append("")
    
    return result
